### PR TITLE
YJIT: Avoid splitting mov for small values on arm64

### PR DESCRIPTION
**Pull request by k0kubun**
When `Mov`'s destination is a register and the source is an immediate which cannot be encoded with a bitmask but is <= `0xffff`, it can be a single `mov` instruction. 

Currently, `split_bitmask_immediate` splits such source operands, which are still necessary for other instructions, so this PR avoids calling it for that case.

While I don't see a significant impact on benchmarks, this should help the performance of stack temp register allocation on arm64 https://github.com/ruby/ruby/pull/7659 a little.

### before
```asm
  # Insn: 0000 putobject (stack_size: 0)
  # reg_temps: 00000000 -> 00000001
  0x1065d0090: mov x11, #5
  0x1065d0094: mov x1, x11
```

### after
```asm
  # Insn: 0000 putobject (stack_size: 0)
  # reg_temps: 00000000 -> 00000001
  0x103840090: mov x1, #5
```

### CODE REVIEW

File: yjit/src/backend/arm64/mod.rs [hunk 1]
```diff
@@ -956,7 +960,18 @@ impl Assembler
                     };
                 },
                 Insn::Mov { dest, src } => {
-                    mov(cb, dest.into(), src.into());
+                    // This supports the following two kinds of immediates:
+                    //   * The value fits into a single movz instruction
+                    //   * It can be encoded with the special bitmask immediate encoding
+                    // arm64_splat should have split other immediates that require multiple instructions.
```
Comment (XrXr) [inline yjit/src/backend/arm64/mod.rs:966]:
```suggestion
                    // arm64_split() should have split other immediates that require multiple instructions.
```

File: yjit/src/backend/arm64/mod.rs [hunk 2]
```diff
@@ -1588,4 +1603,16 @@ mod tests {
             0x0: ldur x1, [x19, #8]
         "});
     }
+
+    #[test]
+    fn test_not_split_mov() {
+        let (mut asm, mut cb) = setup_asm();
+
+        asm.mov(Opnd::Reg(Assembler::TEMP_REGS[0]), Opnd::UImm(5));
```
Comment (XrXr) [inline yjit/src/backend/arm64/mod.rs:1611]: I would test with 0xffff and 0x10000 to show that it handles the boundary condition properly.

**Review by XrXr — APPROVED** [review]
Nice!

Off topic, sometimes our `asm.mov` generates `orr` in the disassembly because I think [for small values `movz` gets the `mov` mnemonic](https://developer.arm.com/documentation/ddi0596/2021-12/Base-Instructions/MOVZ--Move-wide-with-zero-). Might want to change the selection there just for the looks of the disassembly.

**Review by maximecb — APPROVED** [review]

**Comment by k0kubun** [comment]
> Off topic, sometimes our asm.mov generates orr in the disassembly because I think [for small values movz gets the mov mnemonic](https://developer.arm.com/documentation/ddi0596/2021-12/Base-Instructions/MOVZ--Move-wide-with-zero-). Might want to change the selection there just for the looks of the disassembly.

I would like a non-`orr` output too, but is there an instruction that's useful for this? I believe we can't control what libcapstone outputs when multiple instructions have the same encoding, so I'm assuming you're talking about a different instruction, but is that still `mov`-like one?

**Comment by XrXr** [comment]
I'm not very sure what exactly is going on either. It might be wishful thinking that Capstone tries to follow the spec when picking mnemonics. I'll start taking notes when I see orr and not spend dedicated time on this hah.
